### PR TITLE
clang-tidy modernize-use-nullptr

### PR DESCRIPTION
**Pull request by stellar-aria**
Mechanically replaces NULL usage with nullptr

### CODE REVIEW

File: src/deluge/util/functions.cpp [hunk 1]
```diff
@@ -1347,7 +1347,7 @@ bool isAudioFilename(char const* filename) {
 
 bool isAiffFilename(char const* filename) {
 	char* dotPos = strrchr(filename, '.');
-	return (dotPos != 0 && (!strcasecmp(dotPos, ".AIF") || !strcasecmp(dotPos, ".AIFF")));
+	return (dotPos != nullptr && (!strcasecmp(dotPos, ".AIF") || !strcasecmp(dotPos, ".AIFF")));
```
Comment (m-m-adams) [inline src/deluge/util/functions.cpp:1350]: should not be nullptr

Comment (stellar-aria) [reply src/deluge/util/functions.cpp]: Fixed

File: src/deluge/modulation/params/param_manager.cpp [hunk 2]
```diff
@@ -30,12 +30,12 @@
 #include <new>
 
 ParamManager::ParamManager() {
-	summaries[0] = {0};
+	summaries[0] = {nullptr};
```
Comment (m-m-adams) [inline src/deluge/modulation/params/param_manager.cpp:33]: This needs to be zero - it's using the gcc extension that treats {0} as initializing all members to zero (including two uint32s)

Comment (m-m-adams) [reply src/deluge/modulation/params/param_manager.cpp:33]: Or {nullptr, 0, 0}

Comment (stellar-aria) [reply src/deluge/modulation/params/param_manager.cpp:33]: fixed

**Review by sapphire-arches — APPROVED** [review]

**Comment by sapphire-arches** [comment]
Using some `grep` we do have some `NULL`s still in `src/deluge` even after this change, in the following files:

```
src/deluge/deluge.cpp
src/deluge/drivers/uart/uart.c
src/deluge/dsp/fft/fft_config_manager.cpp
src/deluge/dsp/timestretch/time_stretcher.cpp
src/deluge/gui/menu_item/dx/param.cpp
src/deluge/gui/menu_item/patch_cable_strength.cpp
src/deluge/gui/menu_item/sequence/direction.h
src/deluge/gui/ui/browser/browser.cpp
src/deluge/gui/ui/browser/dx_browser.cpp
src/deluge/gui/ui/browser/sample_browser.cpp
src/deluge/gui/ui/browser/slot_browser.h
src/deluge/gui/ui/keyboard/keyboard_screen.cpp
src/deluge/gui/ui/load/load_instrument_preset_ui.cpp
src/deluge/gui/ui/load/load_instrument_preset_ui.h
src/deluge/gui/ui/load/load_song_ui.cpp
src/deluge/gui/ui/save/save_instrument_preset_ui.cpp
src/deluge/gui/ui/ui.cpp
src/deluge/gui/views/arranger_view.cpp
src/deluge/gui/views/automation_view.cpp
src/deluge/gui/views/instrument_clip_view.cpp
src/deluge/gui/views/session_view.cpp
src/deluge/gui/views/view.cpp
src/deluge/hid/display/oled.cpp
src/deluge/hid/led/pad_leds.cpp
src/deluge/io/midi/learned_midi.cpp
src/deluge/io/midi/midi_device_manager.cpp
src/deluge/io/midi/midi_device_manager.h
src/deluge/io/midi/midi_engine.cpp
src/deluge/memory/general_memory_allocator.cpp
src/deluge/memory/memory_region.cpp
src/deluge/model/action/action.h
src/deluge/model/action/action_logger.cpp
src/deluge/model/clip/audio_clip.h
src/deluge/model/clip/clip.cpp
src/deluge/model/clip/instrument_clip.cpp
src/deluge/model/consequence/consequence_clip_existence.cpp
src/deluge/model/drum/drum.h
src/deluge/model/global_effectable/global_effectable.cpp
src/deluge/model/global_effectable/global_effectable_for_clip.cpp
src/deluge/model/instrument/kit.cpp
src/deluge/model/instrument/melodic_instrument.cpp
src/deluge/model/instrument/melodic_instrument.h
src/deluge/model/instrument/midi_instrument.cpp
src/deluge/model/instrument/non_audio_instrument.h
src/deluge/model/mod_controllable/mod_controllable_audio.cpp
src/deluge/model/mod_controllable/mod_controllable_audio.h
src/deluge/model/mod_controllable/mod_controllable.cpp
src/deluge/model/mod_controllable/mod_controllable.h
src/deluge/model/model_stack.cpp
src/deluge/model/model_stack.h
src/deluge/model/note/note_row.cpp
src/deluge/model/sample/sample_cache.cpp
src/deluge/model/sample/sample_cluster.h
src/deluge/model/sample/sample.cpp
src/deluge/model/sample/sample_holder.cpp
src/deluge/model/sample/sample_low_level_reader.cpp
src/deluge/model/sample/sample_playback_guide.h
src/deluge/model/sample/sample_recorder.cpp
src/deluge/model/sample/sample_recorder.h
src/deluge/model/song/song.cpp
src/deluge/model/voice/voice.cpp
src/deluge/model/voice/voice_sample.cpp
src/deluge/modulation/params/param_collection.h
src/deluge/modulation/params/param_manager.cpp
src/deluge/modulation/params/param_manager.h
src/deluge/modulation/patch/patch_cable_set.cpp
src/deluge/modulation/patch/patcher.cpp
src/deluge/processing/audio_output.cpp
src/deluge/processing/engines/audio_engine.cpp
src/deluge/processing/live/live_pitch_shifter.cpp
src/deluge/processing/sound/sound.cpp
src/deluge/processing/sound/sound_drum.cpp
src/deluge/processing/sound/sound.h
src/deluge/processing/sound/sound_instrument.cpp
src/deluge/processing/source.cpp
src/deluge/storage/audio/audio_file_holder.cpp
src/deluge/storage/audio/audio_file_manager.cpp
src/deluge/storage/Deserializer.cpp
src/deluge/storage/JsonDeserializer.cpp
src/deluge/storage/JSONSerializer.cpp
src/deluge/storage/smsysex.cpp
src/deluge/storage/storage_manager.cpp
src/deluge/storage/storage_manager.h
src/deluge/storage/wave_table/wave_table.cpp
src/deluge/testing/automated_tester.cpp
src/deluge/util/container/list/bidirectional_linked_list.h
src/deluge/util/d_string.h
src/deluge/util/functions.cpp
```

A few of those are comment-only, but the majority are not.

**Comment by stellar-aria** [comment]
@sapphire-arches I used clang-tidy, so it probably only changed the ones it was sure of, and even then I kept some NULLs as character terminators for strings and stuff

**Review by sapphire-arches — APPROVED** [review]

**Pull request merged**